loading, width and height attributes are duplicated in the fallback i…

sashau opened the pull request:
…mg tag

The fallback image that is created from the buildSrcset function looks like this:
`<img loading="lazy" width="400" height="300"src="/media/cached-resp-images/images/..." loading="lazy" width="400" height="300">`

Removing this line the fallback image becomes:
`<img src="/media/cached-resp-images/images/..." loading="lazy" width="400" height="300">`

The loading, width and height attributes should already come from the joomla.html.image layout.

sashau committed:
loading, width and height attributes are duplicated in the fallback img tag

The fallback image that is created from the buildSrcset function looks like this:
<img loading="lazy" width="400" height="300"src="/media/cached-resp-images/images/..." loading="lazy" width="400" height="300">

Removing this line the fallback image becomes:
<img src="/media/cached-resp-images/images/..." loading="lazy" width="400" height="300">

The loading, width and height attributes should already come from the joomla.html.image layout.

dgrammatiko commented:
@sashau thanks for the PR, this is a obviously a bug, but I would like to have the values for both attributes from the cached data instead from the original tag. (has to do with cache invalidation and the case that an image could be edited, eg resized, thus the original attributes could be misleading).

sashau committed:
Used cached width/height for fallback image

I had a go at using the cached width and height for the fallback image. The first part replaces any width/height attributes (along with the src) with the cached values and the second part adds in width/height/loading if they don't exist in the original image.

sashau commented:
I had a go at using the cached width and height for the fallback image. The first part replaces any width/height attributes (along with the src) with the cached values and the second part adds in width/height/loading if they don't exist in the original image.

dgrammatiko commented:
It's perfect, thank you @sashau
